main.tf

```
@@ -0,0 +1,111 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source = "hashicorp/azurerm"
+      version = ">= 3.8"
+    }
+  }
+  backend "azurerm" {
+    resource_group_name  = "Cohort28_RajSal_ProjectExercise"
+    storage_account_name = "rsm12st01"
+    container_name       = "rsm12st01container"
+    key                  = "terraform.tfstate"
+    }
+}
+
+provider "azurerm" {
+  features {}
+}
+
+data "azurerm_resource_group" "main" {
+  name     = "Cohort28_RajSal_ProjectExercise"
+}
+
+# AppServicePlan
+resource "azurerm_service_plan" "main" {
+  name                = "terraformed-asp"
+  location            = data.azurerm_resource_group.main.location
+  resource_group_name = data.azurerm_resource_group.main.name
+  os_type             = "Linux"
+  sku_name            = "B1"
+}
+
+# AppService
+resource "azurerm_linux_web_app" "main" {
+  name                = "rs-m12-app"
+  location            = data.azurerm_resource_group.main.location
+  resource_group_name = data.azurerm_resource_group.main.name
+  service_plan_id     = azurerm_service_plan.main.id
+
+  site_config {
+    application_stack {
+      docker_image     = "rsalhan/m8-todo-app"
+      docker_image_tag = "prod"
```

You might see a deprecation warning around these variables - the newer required syntax is:
```suggestion
      docker_image_name = "rsalhan/m8-todo-app:prod"
      docker_registry_url   = "https://index.docker.io"
```

---

Cheers Jack, new syntax has now been included.